### ISSUE TITLE
Source Maps for :npm-module targets missing offset

### ISSUE THREAD

**Issue by aisamu** [thheller/shadow-cljs#617]
(From clojurians `#shadow-cljs`) 

Source maps for `:npm-module` targets (development) appear to not be taking into account the shimming preamble. 
Deleting that and inspecting the maps with source-map-visualization[1] works as expected.

Here's an example of what I'm calling a preamble:
```js
var $CLJS = require("./cljs_env");
var $jscomp = $CLJS.$jscomp;
require("./cljs.core.js");
var module$shadow_js_shim_module$$material_ui$styles$ThemeProvider=$CLJS.module$shadow_js_shim_module$$material_ui$styles$ThemeProvider || ($CLJS.module$shadow_js_shim_module$$material_ui$styles$ThemeProvider = {});
[...]
var shadow=$CLJS.shadow || ($CLJS.shadow = {});
[...]
var module$shadow_js_shim_module$$material_ui$icons$StarHalfRounded=$CLJS.module$shadow_js_shim_module$$material_ui$icons$StarHalfRounded || ($CLJS.module$shadow_js_shim_module$$material_ui$icons$StarHalfRounded = {});
// empty line
$CLJS.SHADOW_ENV.setLoaded("cljs.namespace.js");
// empty line
```
The development source maps are being processed by Webpack's `source-map-loader`, and appear to be consumed correctly by the browser. (i.e. the browser jumps to the correct Cljs file, but lands on an incorrect line/column)
 
I couldn't check release builds - the tool complained about missing sources.

Source maps for `:browser-test` worked out of the box!

[1] - https://sokra.github.io/source-map-visualization/#custom

**Comment by aisamu** [comment]
🎉 
Wow, that was fast - thanks a lot!

**Comment by thheller** [comment]
Released in `2.8.82`.
